src/hardware/opl.cpp

```diff
@@ -31,254 +31,294 @@
+			fseek(handle, 0, SEEK_SET);
+			fwrite(&header, 1, sizeof(header), handle);
+			fclose(handle);
 			handle = 0;
 		}
 	}
+
 public:
-	bool DoWrite( uint32_t regFull, uint8_t val ) {
+	bool DoWrite(uint32_t regFull, uint8_t val)
+	{
 		uint8_t regMask = regFull & 0xff;
-		//Check the raw index for this register if we actually have to save it
-		if ( handle ) {
+		// Check the raw index for this register if we actually have to
+		// save it
+		if (handle) {
 			/*
-				Check if we actually care for this to be logged, else just ignore it
+			        Check if we actually care for this to be logged,
+			   else just ignore it
 			*/
-			uint8_t raw = ToRaw[ regMask ];
-			if ( raw == 0xff ) {
+			uint8_t raw = ToRaw[regMask];
+			if (raw == 0xff) {
 				return true;
 			}
-			/* Check if this command will not just replace the same value 
-			   in a reg that doesn't do anything with it
+			/* Check if this command will not just replace the same
+			   value in a reg that doesn't do anything with it
 			*/
-			if ( (*cache)[ regFull ] == val )
+			if ((*cache)[regFull] == val)
 				return true;
 			/* Check how much time has passed */
 			uint32_t passed = PIC_Ticks - lastTicks;
-			lastTicks = PIC_Ticks;
+			lastTicks       = PIC_Ticks;
 			header.milliseconds += passed;
 
-			//if ( passed > 0 ) LOG_MSG( "Delay %d", passed ) ;
-			
-			// If we passed more than 30 seconds since the last command, we'll restart the the capture
-			if ( passed > 30000 ) {
+			// if ( passed > 0 ) LOG_MSG( "Delay %d", passed ) ;
+
+			// If we passed more than 30 seconds since the last
+			// command, we'll restart the the capture
+			if (passed > 30000) {
 				CloseFile();
-				goto skipWrite; 
+				goto skipWrite;
 			}
 			while (passed > 0) {
-				if (passed < 257) {			//1-256 millisecond delay
-					AddBuf( delay256, passed - 1 );
+				if (passed < 257) { // 1-256 millisecond delay
+					AddBuf(delay256, passed - 1);
 					passed = 0;
 				} else {
 					const auto shift = (passed >> 8);
 					passed -= shift << 8;
-					AddBuf( delayShift8, shift - 1 );
+					AddBuf(delayShift8, shift - 1);
 				}
 			}
-			AddWrite( regFull, val );
+			AddWrite(regFull, val);
 			return true;
 		}
-skipWrite:
-		//Not yet capturing to a file here
-		//Check for commands that would start capturing, if it's not one of them return
-		if ( !(
-			//note on in any channel 
-			( regMask>=0xb0 && regMask<=0xb8 && (val&0x020) ) ||
-			//Percussion mode enabled and a note on in any percussion instrument
-			( regMask == 0xbd && ( (val&0x3f) > 0x20 ) )
-		)) {
+	skipWrite:
+		// Not yet capturing to a file here
+		// Check for commands that would start capturing, if it's not
+		// one of them return
+		if (!(
+		            // note on in any channel
+		            (regMask >= 0xb0 && regMask <= 0xb8 && (val & 0x020)) ||
+		            // Percussion mode enabled and a note on in any
+		            // percussion instrument
+		            (regMask == 0xbd && ((val & 0x3f) > 0x20)))) {
```

Comment:
`clang-format` struggled with this one!
maybe temps would help.. 
``` c++
const auto is_note_on_any_ch = regMask >= 0xb0 && regMask <= 0xb8 && (val & 0x020);
const auto is_perc_on_with_note = regMask == 0xbd && ((val & 0x3f) > 0x20));
if (!is_note_on_any_ch && !is_perc_on_with_note)
	return;
```
(or something a bit more digestible)

Reply:
😅 I agree